spec/controllers/products_controller_spec.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ProductsController, type: :controller do
+  let(:category) { create(:category) }
+  let!(:product) { create(:product, category:) }
+  let(:serialized_product) { ProductSerializer.new(product).serializable_hash }
+  let(:error_message) { 'Product not found' }
+
+  let(:service_result) { instance_double('ServiceResult', success?: true, result: product) }
+  let(:service_failure) do
+    instance_double('ServiceFailure', success?: false, first_error: instance_double('Error', message: error_message))
+  end
+
+  describe 'GET #show' do
+    context 'when the UUID format is invalid' do
```

Comment:
context dependency, what if in future id is not UUID anymore, when params is invalid